cmd/libs/go2idl/client-gen/generators/generator-for-clientset.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"k8s.io/kubernetes/cmd/libs/go2idl/generator"
+	"k8s.io/kubernetes/cmd/libs/go2idl/namer"
+	"k8s.io/kubernetes/cmd/libs/go2idl/types"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// genClientset generates a package for a clientset.
+type genClientset struct {
+	generator.DefaultGen
+	groupVersions   []unversioned.GroupVersion
+	typedClientPath string
+	outputPackage   string
+	imports         *generator.ImportTracker
+}
+
+func (g *genClientset) Namers(c *generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.outputPackage, g.imports),
+	}
+}
+
+var generate_clientset = true
+
+// We only want to call GenerateType() once.
+func (g *genClientset) Filter(c *generator.Context, t *types.Type) bool {
+	ret := generate_clientset
+	generate_clientset = false
+	return ret
+}
+
+func normalizeGroup(group string) string {
+	if group == "api" {
+		return "legacy"
+	}
+	return group
+}
+
+func normalizeVersion(version string) string {
+	if version == "" {
+		return "unversioned"
+	}
+	return version
+}
+
+func (g *genClientset) Imports(c *generator.Context) (imports []string) {
+	for _, gv := range g.groupVersions {
+		group := normalizeGroup(gv.Group)
+		version := normalizeVersion(gv.Version)
+		typedClientPath := filepath.Join(g.typedClientPath, group, version)
+		imports = append(imports, g.imports.ImportLines()...)
+		imports = append(imports, fmt.Sprintf("%s_%s \"%s\"", group, version, typedClientPath))
+	}
+	return
+}
+
+// TODO: We actually don't need any type information to generate the clientset,
```

Comment:
It will then look like a dangling comment :)

I am fine with keeping this as is if that is what you want. But I would have moved the comment inside the function.